core/eventticker/eventticker_test.go

```diff
@@ -0,0 +1,81 @@
+package eventticker_test
+
+import (
+	"encoding/binary"
+	"testing"
+	"time"
+
+	"github.com/iotaledger/hive.go/core/eventticker"
+	"github.com/mr-tron/base58"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventTicker(t *testing.T) {
+	eventTicker := eventticker.New[index, testID]()
+
+	id := newtestID(index(1), 0)
+	eventTicker.StartTicker(id)
+
+	has := eventTicker.HasTicker(id)
+	require.True(t, has, "EventTicker.HasTicker() returned false, expected true")
+	require.Equalf(t, 1, eventTicker.QueueSize(), "EventTicker.QueueSize() returned %d, expected 1", eventTicker.QueueSize())
+
+	eventTicker.StopTicker(id)
+	has = eventTicker.HasTicker(id)
+	require.False(t, has, "EventTicker.HasTicker() returned true, expected false")
+	require.Equalf(t, 0, eventTicker.QueueSize(), "EventTicker.QueueSize() returned %d, expected 0", eventTicker.QueueSize())
+
+	id1 := newtestID(index(1), 1)
+	id2 := newtestID(index(2), 2)
+	eventTicker.StartTickers([]testID{id1, id2})
+
+	has = eventTicker.HasTicker(id1)
+	require.True(t, has, "EventTicker.HasTicker() returned false, expected true")
+	has = eventTicker.HasTicker(id2)
+	require.True(t, has, "EventTicker.HasTicker() returned false, expected true")
+
+	eventTicker.EvictUntil(index(1))
+	require.Equalf(t, 1, eventTicker.QueueSize(), "EventTicker.QueueSize() returned %d, expected 1", eventTicker.QueueSize())
+
+	done := make(chan struct{})
```

Comment:
Run the test a little longer to make sure that the function is actually executed multiple times at the correct time interval.